org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/MBEventHandler.java

```diff
@@ -157,8 +222,518 @@ public void handleEvent(final Event event) {
+
+    private Map<String, String> toEndpointMetadata(Collection<String> sourceNodeIDs,
+                                                   Collection<String> sourceRelationIDs,
+                                                   Collection<String> targetNodeIDs,
+                                                   Collection<String> targetRelationIDs) {
+        Map<String, String> result = new HashMap<String, String>();
+
+        result.put("SOURCENODES", this.toCSV(sourceNodeIDs));
+        result.put("SOURCERELATIONS", this.toCSV(sourceRelationIDs));
+        result.put("TARGETNODES", this.toCSV(targetNodeIDs));
+        result.put("TARGETRELATIONS", this.toCSV(targetRelationIDs));
+
+        return result;
+    }
+
+    private Collection<String> toStringCollection(String data, String separator) {
+        Collection<String> result = new ArrayList<String>();
+
+        if(data == null || data.isEmpty()) {
+            return result;
+        }
+        
+        String[] split = data.split(separator);
+
+        for (String part : split) {
+            if (part != null && !part.equals("") && !part.isEmpty()) {
+                result.add(part);
+            }
+        }
+
+        return result;
+    }
+
+    private String toCSV(Collection<String> strings) {
+        StringBuilder strB = new StringBuilder();
+
+        strings.forEach(s -> strB.append(s).append(","));
+
+        return strB.toString();
+    }
+
+    private Set<PlanInstanceInput> toPlanInstanceInputs(Map<String, String> inputs) {
+        Set<PlanInstanceInput> result = new HashSet<PlanInstanceInput>();
+
+        for (String key : inputs.keySet()) {
+            String val = inputs.get(key);
+            result.add(new PlanInstanceInput(key, val, "string"));
+        }
+
+        return result;
+    }
+
+    private Map<String, String> createInput(BPELPlan plan) {
+        Collection<String> inputs = plan.getWsdl().getInputMessageLocalNames();
+
+        Map<String, String> result = new HashMap<String, String>();
+
+        for (String input : inputs) {
+            result.put(input, null);
+        }
+
+        return result;
+    }
+
+    private BpelPlanEnginePlugin getBpelDeployPlugin() {
+        BundleContext ctx = Activator.ctx;
+        try {
+            ServiceReference<?>[] refs =
+                ctx.getAllServiceReferences(IPlanEnginePlanRefPluginService.class.getName(), null);
+            if (refs != null) {
+                for (ServiceReference<?> ref : refs) {
+                    IPlanEnginePlanRefPluginService plugin = (IPlanEnginePlanRefPluginService) ctx.getService(ref);
+                    if (plugin instanceof BpelPlanEnginePlugin) {
+                        return (BpelPlanEnginePlugin) plugin;
+                    }
+                }
+            }
+        }
+        catch (InvalidSyntaxException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+            return null;
+        }
+        return null;
```

Comment:
There should be a `PluginRegistry` around that can take care of that for you...

Reply:
Don't know if you mean the PluginRegistry of the PlanBuilder but I don't find such a thing in the bus, and at that point of the code I just need the endpoint service